### ISSUE TITLE
Add support for LuaSnip or snippets.nvim

### ISSUE THREAD

**Issue by doums** [ms-jpq/coq_nvim#145]
Hi,

It would be great if you add support for snippet plugins dedicated to neovim (and so supporting lua) like [LuaSnip](https://github.com/L3MON4D3/LuaSnip) and/or [snippets.nvim](https://github.com/norcalli/snippets.nvim).

**Comment by ms-jpq** [comment]
sorry, thats out of scope, maintaining compatibility with third party plugins that can change their API anytime is too much work for me to realistically do..

**Comment by ms-jpq** [comment]
I'd rather put in that time to improve other parts of `coq`, including the snippet system

**Comment by samrathchadha** [comment]
Is there any possibility for user-defined snippets then? If that is possible, I could write a plugin that somehow extends coq and adds use with LuaSnip and snippets.nvim.

**Comment by ms-jpq** [comment]
see the issue here.

https://github.com/ms-jpq/coq_nvim/pull/168

---

tldr:

built-in support is coming, but its gonna take some time because i want to make it good, like really good
